crates/h3arrow/Cargo.toml

```diff
@@ -21,8 +21,7 @@ spatial_index = ["dep:rstar"]
 [dependencies]
 ahash = "0.8"
 arrow = { workspace = true }
-# geoarrow = { package = "geoarrow", version = "0.1.0", optional = true, features = ["geozero"] }
-geoarrow = { package = "geoarrow", git = "https://github.com/geoarrow/geoarrow-rs.git", rev = "30c18f7c07432dfb92f9612efa54aec1d9392daf", optional = true, features = ["geozero"] }
+geoarrow = { version = "0.3", optional = true, features = ["geozero"] }
```

Comment:
I just published 0.4.0-beta.1 if you want to bump all the way up to that. You'll just have to take off the `O` parameter.

But note there's also now a separation between "native" GeoArrow arrays and "serialized" GeoArrow arrays. WKB is a "serialized" array.

If you get stuck I'm happy to help update this code

Reply:
I am bit out of the loop on what is happening in the arrow-community. I remember the "native" geoarrow arrays where a bit problematic as there was no way to attach the metadata describing the geometrytype to the array. Is this now possible? Is it feasible to exchange the "native" arrays with other libraries?

Reply:
No, that's not what I mean. I mean the GeoArrow format has both "Native encoding" and "Serialized encoding" https://geoarrow.org/format.html.

Previously in geoarrow-rs we had a single `GeometryArrayTrait` that all geoarrow array types implemented, but now we have two:

- [`NativeArray`](https://docs.rs/geoarrow/0.4.0-beta.1/geoarrow/trait_/trait.NativeArray.html), which is implemented by `PointArray`, `LineStringArray`, `PolygonArray`, `MultiPointArray`, `MultiLineStringArray`, `MultiPolygonArray`, `MixedGeometryArray`, and `GeometryCollectionArray`.
- [`SerializedArray`](https://docs.rs/geoarrow/0.4.0-beta.1/geoarrow/trait_/trait.SerializedArray.html), which is implemented by `WKBArray` and `WKTArray`.

I just mean that you'll have to update the code to use these new traits

Reply:
Ok, now I got it ;)

BTW: Directly upgrading to geoarrow 0.4.0-beta.1 is not possible for me currently. This requires at least he release of rust-numpy built ontop of pyo3 0.22

Reply:
I'm also hoping that will come out very soon. https://github.com/PyO3/rust-numpy/pull/442#issuecomment-2384038506 and https://github.com/PyO3/rust-numpy/pull/435 have been merged, which is very promising